hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMFailoverProxyProviderBase.java

```diff
@@ -172,12 +175,23 @@ public RetryAction shouldRetry(Exception exception, int retries,
           OMNotLeaderException notLeaderException =
               getNotLeaderException(exception);
           if (notLeaderException != null) {
-            // TODO: NotLeaderException should include the host
-            //  address of the suggested leader along with the nodeID.
-            //  Failing over just based on nodeID is not very robust.
-
             // Prepare the next OM to be tried. This will help with calculation
             // of the wait times needed get creating the retryAction.
+            if (notLeaderException.getSuggestedLeaderAddress() != null &&
+                notLeaderException.getSuggestedLeaderNodeId() != null) {
+              String suggestedLeaderAddress =
+                  notLeaderException.getSuggestedLeaderAddress();
+              String suggestedNodeId =
+                  notLeaderException.getSuggestedLeaderNodeId();
+              if (omNodeAddressMap.containsKey(suggestedNodeId) &&
```

Comment:
_Instead_ of calling `notLeaderException.getSuggestedLeaderAddress() `and `getSuggestedLeaderNodeId() ` twice, first checking if non-null then setting variables `suggestedLeaderAddress` and `suggestedNodeID`, _we can just_ call once setting the local variables and check if variables are non-null later before `setNextOmProxy()`.